aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -1658,101 +1666,120 @@ Tensor _matmul_impl(
     return has_out ? at::mv_out(out, tensor2.t(), tensor1) : tensor2.t().mv(tensor1);
   } else if (dim_tensor1 == 2 && dim_tensor2 == 2) {
     return has_out ? at::mm_out(out, tensor1, tensor2) : tensor1.mm(tensor2);
-  } else if (should_fold_into_mm(tensor1, tensor2)) {
+  } else if (should_fold_into_mm(tensor1, tensor2) ||
+             should_fold_into_mm(tensor2, tensor1)) {
     // optimization: use mm instead of bmm by folding tensor1's batch into
```

Comment:
This is technically inaccurate now as we may also fold tensor2's batch

Reply:
Note that this was being done in an else that would call this function recursively with the transposed arguments. We merge the two to avoid the second function call and having to recompute things.